core/js/src/main/scala/porcupine/dbplatform.scala

```diff
@@ -52,7 +52,12 @@ private abstract class DatabasePlatform:
                       def cursor(args: A): Resource[F, Cursor[F, B]] = mutex.lock *>
                         Resource
                           .eval {
-                            F.delay(statement.iterate(bind(args)*)).map { iterator =>
+                            val argsList = bind(args)
+                            val argsDict: js.Dictionary[Any] =
+                              js.Dictionary(argsList.zipWithIndex.map { case (v, idx) =>
+                                (s"${idx + 1}", v)
+                              }*)
+                            F.delay(statement.iterate(argsDict)).map { iterator =>
```

Comment:
Let's put this all in the `delay` block, and I think we can build the JS object more efficiently without all the intermediate datastructures form `zipWithIndex` and `map`. Instead, we can just create a new `js.Object` and set its keys with `js.Dynamic`.